content/blog/2018-08-28-birds-ocr.md

```diff
@@ -0,0 +1,411 @@
+title: "What's this bird? Classify old natural history drawings with R"
+slug: birds-ocr
+authors:
+  - name: Maëlle Salmon
+    url: https://masalmon.eu/
+date: 2018-08-28
+preface: The blog post series corresponds to the material for a talk Maëlle will give at the [Animal Movement Analysis summer school in Radolfzell, Germany on September the 12th](http://animove.org/animove-2019-evening-keynotes/), in a Max Plant Institute of Ornithology.
+tags:
+- magick
+- tesseract
+- cld2
+- cld3
+- taxize
+- birder
+output: 
+  md_document:
+    variant: markdown_github
+    preserve_yaml: true
+---
+
+In this new post, we’re taking a break from modern birding data in [our
+birder’s series](https://ropensci.org/tags/birder)… let’s explore
+gorgeous drawings from a natural history collection! Armed with
+rOpenSci’s packages binding powerful C++ libraries and open taxonomy
+data, how much information can we automatically extract from images?
+Maybe not much, but we’ll at least have explored image manipulation,
+optical character recognition, language detection, taxonomic name
+resolution with rOpenSci’s packages.
+
+### Free natural history images and appropriate R tooling!
+
+A long time ago I had bookmarked the [Flickr account of the Biodiversity
+Heritage Library](https://www.flickr.com/people/biodivlibrary/). So many
+beautiful images of biodiversity, moreover free to use! In particular, I
+[downloaded all pictures from one of the Birds of Australia
+albums](https://www.flickr.com/photos/biodivlibrary/sets/72157694191194992).
+
+I wanted to try to extract the bird names on images using packages of
+Jeroen Ooms’, rOpenSci’s post-doc hacker & C(++)-bindings wizard. In
+particular, I wanted to try extract names of
+[`magick`](https://github.com/ropensci/magick) for image manipulation,
+[`tesseract`](https://github.com/ropensci/tesseract) for optical
+character recognition,
+[`cld2`](https://github.com/ropensci/cld2)/[`cld3`](https://github.com/ropensci/cld3)
+for language detection… Quite the armory! We’ll also sprinkle some
+taxonomy magic by Scott Chamberlain, one of rOpenSci’s co-founders, to
+resolve the names extracted.
+
+### OCR-bird naming workflow, piece by piece
+
+In this section, we explain the different elements of our R workflow:
+preparing images, extracting text, resolving taxonomic names.
+
+#### Image preparation
+
+I saved the pictures locally in a “birds” folder. Yes, I click-buttoned
+instead of using [the Flickr API](https://www.flickr.com/services/api/)
+for which e.g. Jim Hester wrote a [minimal R
+wrapper](https://github.com/jimhester/flickrr)… I don’t do *everything*
+with R scripts (yet).
+
+``` r
+library("magrittr")
+filenames <- fs::dir_ls("birds")
+```
+
+Each image shows a bird and its name. Images are either landscape- or
+portrait- oriented, but for the sake of simplicity, I’ll act as if they
+were all portrait- oriented. *What I’d do if I weren’t after simplicity
+but still as lazy would be to duplicate images rotated
```

Comment:
revised for reader clarity:
*What I’d do if I weren’t after simplicity
but still as lazy, would be to duplicate images rotated
(`magick::image_rotate`) in all possible directions and apply the
workflow, hoping to get good data from one of the rotated versions.*

Reply:
rephrased